R/stroke.R

```diff
@@ -130,38 +130,40 @@ get_linked_nodes <- function(node_id, segment_id, segments) {
 
 #' @noRd
 best_link <- function(nodes, segments, links, angle_threshold = 0) {
+    
+    # convert nodes to a matrix for faster indexing
+    nodes <- as.matrix(nodes[c("x", "y")])
+    
+    best_links <- array(integer(), dim = dim(segments))
+    colnames(best_links) <- c("start", "end")
+    
+    angle_threshold_rad <- angle_threshold / 180 * pi  # convert to radians
+    
+    for (iseg in seq_len(nrow(segments))) {
+        start_node <- segments[iseg, "start"]
+        end_node <- segments[iseg, "end"]
+        
+        # Use the helper function to determine best link at start and end nodes
+        best_link_start <- find_best_link(start_node, end_node, iseg, segments, links)
```

Comment:
This function also requires `nodes` and `angle_threshold`:

```suggestion
        best_link_start <- find_best_link(start_node, end_node, iseg, segments, links, nodes, angle_threshold)
```